### PR TITLE
Fix players seeing lights being toggled on or off at roundstart

### PR DESCRIPTION
**Pull request by Hinaichigo**
## What this does
This makes it so all the light, lightswitch, and lamp on-or-off toggling that depends on whether or not the department is occupied at roundstart, occurs before players are transferred into their bodies, so that the player won't see the lights changing right at the start of the round. To get this to work, these things are done before body transfer occurs:
- lightswitches, lamps, and lightbulbs themselves are force-updated 
- a complete tick of the lighting subsystem is forced to fire, updating the lighting overlays

This is a remake of #35143 without changing which rooms are lit at roundstart, so here all the rooms in a department will have the lights on as long as a crewmember spawns in one of them, so there's no change in that regard.

## Why it's good
Hopefully fix the longstanding issues relating to lights being on-or-off at roundstart.

## Changelog
:cl:
 * bugfix: Fixed being able to see lights toggling at roundstart based on crew occupancy.

### CODE REVIEW

File: code/game/gamemodes/gameticker.dm [hunk 1]
```diff
@@ -648,18 +657,15 @@ var/datum/controller/gameticker/ticker
...
-		for(var/obj/machinery/light_switch/LS in DA)
-			LS.toggle_switch(0, playsound = FALSE)
-			break
-		for(var/obj/item/device/flashlight/lamp/L in DA)
-			L.toggle_onoff(0)
+/datum/controller/gameticker/proc/handle_lights(list/areas_to_turn_lights_on)
+	for(var/area/this_area in areas_to_turn_lights_on)
+		for(var/obj/machinery/light_switch/LS in this_area)
+			LS.toggle_switch(1, playsound = FALSE)
+		for(var/obj/machinery/light/lightykun in this_area)
```
Comment (Kurfursten) [inline code/game/gamemodes/gameticker.dm:664]: going from var names like discrete_areas to lightykun

the west has fallen

**Comment by Kurfursten** [comment]
Code looks fine with a rough skim, did you conduct testing as a few roles? Force yourself as a wizard or nuke op. Any time you touch player spawning it is critical to make sure we don't disrupt gameplay setup.

**Comment by Hinaichigo** [comment]
I'll check those.

**Comment by Hinaichigo** [comment]
> Force yourself as a wizard or nuke op.

How do I do this? I tried becoming a ghost and using "create antagonist" but I don't count as a potential candidate seemingly.

**Comment by Kurfursten** [comment]
Use the Game Panel (admin tab) to Force Roundstart Mode, then make sure your preferences are enabled and ready up, then use the special menu to Start Now

We don't want to use a midround "create antagonist" anyway, we want to make sure they're spawning correct at roundstart

**Comment by Hinaichigo** [comment]
On Bleeding-Edge to check first, I did this:

- set my prefs to Always for all antag roles
- set the game mode to Wizard with the Game Panel
- Readied
- pressed Start Now

It says I'm now a wizard but I stay in the start area as a ghost mob.
I think I'm doing something wrong.

**Comment by Kurfursten** [comment]
Try multiboxing (set your config to allow guests) and ready up in the other window. It may be rejecting game start due to 0 crewmen.

**Comment by Hinaichigo** [comment]
Even with a guest account also logged in and ready, the non-wizard spawns normally but the wizard doesn't spawn...
I'm not sure.

One thing is that I didn't change anything with regard to spawn location stuff (except for making silicons only transfer their new body to the station while the new_player mob stays at the start location, but I tested that and it works). Mainly this just changes the location and logic of the light handling step.

**Comment by Hinaichigo** [comment]
Oh so it seems to be caused by a runtime in the wizard spawning (testing in Bleeding-Edge):
```
Runtime in code/datums/gamemode/role/wizard.dm, line 42: Cannot modify null.flavor_text.
proc name: OnPostSetup (/datum/role/wizard/OnPostSetup)
src: wizard (/datum/role/wizard)
call stack:
wizard (/datum/role/wizard): OnPostSetup()
Wizard Federation (/datum/faction/wizard): OnPostSetup()
Wizard Federation (/datum/faction/wizard): OnPostSetup()
Wizard (/datum/gamemode/wizard): PostSetup()
/datum/controller/gameticker (/datum/controller/gameticker): setup()
/datum/controller/gameticker (/datum/controller/gameticker): pregame()
Ticker (/datum/subsystem/ticker): Initialize(215423)
```
also with nuke ops:
```
Runtime in code/datums/gamemode/factions/syndicate/nukeops.dm,78: Cannot modify null.real_name.
  proc name: OnPostSetup (/datum/faction/syndicate/nuke_op/OnPostSetup)
  src: Syndicate Nuclear Operatives (/datum/faction/syndicate/nuke_op)
  call stack:
  Syndicate Nuclear Operatives (/datum/faction/syndicate/nuke_op): OnPostSetup()
  Nuclear Emergency (/datum/gamemode/nuke): PostSetup()
  /datum/controller/gameticker (/datum/controller/gameticker): setup()
  /datum/controller/gameticker (/datum/controller/gameticker): pregame()
  Ticker (/datum/subsystem/ticker): Initialize(216692)
```

**Comment by Kurfursten** [comment]
It's strange that you're encountering these on bleeding edge because we aren't having problems on live.

Nuke ops L76 - mind.current is null for you, meaning the mind doesn't have a body yet.
Wizard L42 - Again, this is mind.current, which seems to be related to lacking a body for the mind

It's strange because your PR is specifically editing the order in which minds get put in a body. You're sure you're on bleeding edge?

**Comment by Hinaichigo** [comment]
It doesn't edit the order, instead just moves the light handling up to be in-between the body creation and mind/ckey insertion.

I'll test to confirm in a little bit.

**Comment by Hinaichigo** [comment]
@Kurfursten I checked again to be sure with Bleeding-Edge and the same runtime occurs. Maybe it's something with my config, or maybe something that got bugged over time but not noticed since usually the gamemode is Dynamic+.

**Comment by Hinaichigo** [comment]
Conflicts.

**Pull request closed without merging**